src/models/Order.js

```diff
@@ -17,14 +17,19 @@ const orderItemSchema = new mongoose.Schema({
 });
 
 // Schema for the main Order model
-const orderSchema = new mongoose.Schema(
+const OrderSchema = new mongoose.Schema(
   {
     // The user field references the User model and is required
     user: {
       type: mongoose.Schema.Types.ObjectId,
       ref: 'User', // Referencing the User model
       required: [true, 'Please provide user'],
     },
+    email: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'User', // Referencing the User model
+      required: [true, 'Please provide email'],
+    },
```

Comment:
Since we're already referencing the user on line `23-27`, I'm not sure that we need this new field here. We could instead just populate the user field with it's email when we fetch the order?

Eg:

```javascript
Order.findById(orderId).populate({path: 'user', options: { select: { password: 0 } } })
```

Which will return an order object, with an `order.user` field that has all the user info except for the user's password.